extensions/pkg/controller/common/watchdogstarter.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"net"
+	"time"
+
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// DefaultWatchdogInterval is the default watchdog interval used by StartOwnerCheckWatchdog.
+	DefaultWatchdogInterval = 1 * time.Minute
+)
+
+var (
+	// DefaultResolver is the default resolver used by StartOwnerCheckWatchdog.
+	DefaultResolver Resolver = net.DefaultResolver
```

Comment:
I don't think `net.DefaultResolver` caches anything. As discussed, I have reworked the watchdogs so that they are shared per namespace. In addition, they will only perform the check if:
* A result has been requested, and the latest check result is older than 30 seconds.
* Every 30 seconds, but only if there are contexts waiting to be cancelled.
As mentioned, this results in a significantly more complex implementation that I still need to test properly.

Reply:
I tested the new implementation thoroughly and I think it's ok now.